release/tigervnc.iss.in

```diff
@@ -18,6 +18,11 @@ DefaultGroupName=TigerVNC
 #endif
 LicenseFile=@CMAKE_SOURCE_DIR@\LICENCE.txt
 
+[Dirs]
+; This directory is necessary to prevent the X509 file chooser from causing
+; an error dialog to appear when GetOpenFileName is called by SYSTEM account.
+Name: "{sys}\config\systemprofile\Desktop"
+
```

Comment:
-1

We don't want to create random directories all over the place. Can't we handle the error where `GetOpenFileName()` is called?

And why is that running as SYSTEM? It should be running as some form of normal user.

Reply:
I agree that this is ugly, but AFAICT there is no way around it.  From the MS OPENFILENAME struct docs:

'''
Windows 7:

    1. If lpstrInitialDir has the same value as was passed the first time the application used an Open or Save As dialog box, the path most recently selected by the user is used as the initial directory.
    2. Otherwise, if lpstrFile contains a path, that path is the initial directory.
    3. Otherwise, if lpstrInitialDir is not NULL, it specifies the initial directory.
    4. If lpstrInitialDir is NULL and the current directory contains any files of the specified filter types, the initial directory is the current directory.
    5. Otherwise, the initial directory is the personal files directory of the current user.
    6. Otherwise, the initial directory is the Desktop folder.
'''

It seems that GetOpenFileName checks for the existence of the Desktop folder, even if one of the preceding 5 conditions is true, and throws up an error dialog regarding the non-existent directory.  I'm open to suggestions, but I was not able to find any other way to suppress the error dialog.

It runs under the SYSTEM account in service mode.

Reply:
This is part of vncconfig, right? Isn't that started as a more normal user and sends info back to the service? AFAIK trying to use the GUI as a service is a bad idea.

Reply:
Hmm.  I didn't do anything outside of running the installer, so if vncconfig is running under the service account it's because winvnc is setup to do that.

Reply:
Urgh. According to Microsoft's documentation this shouldn't be possible as of Vista, but it seems to work anyway. I guess it is very unsupported though, which is why we get issues like this.

Reply:
Yeah this seems to be a complete nightmare. I'll reluctantly approve of this hack for now. But I'd like a new issue with the fact that we shouldn't run the GUI as SYSTEM, and a reference to that issue in tigervnc.iss.in.